botorch/acquisition/multi_objective/max_value_entropy_search.py

```diff
@@ -175,3 +182,196 @@ def forward(self, X: Tensor) -> Tensor:
+            num_samples: The number of Monte Carlo samples for the Monte Carlo
+                estimate.
+        """
+        super().__init__(
+            model=model,
+            pareto_sets=None,
+            pareto_fronts=None,
+            hypercell_bounds=hypercell_bounds,
+            X_pending=X_pending,
+            estimation_type=estimation_type,
+            num_samples=num_samples,
+        )
+
+    def _compute_posterior_statistics(
+        self, X: Tensor
+    ) -> dict[str, Union[GPyTorchPosterior, Tensor]]:
+        r"""Compute the posterior statistics.
+
+        Args:
+            X: A `batch_shape x q x d`-dim Tensor of inputs.
+
+        Returns:
+            A dictionary containing the posterior variables used to estimate the
+            entropy.
+
+            - "initial_entropy": A `batch_shape`-dim Tensor containing the entropy of
+                the Gaussian random variable `p(Y| X, D_n)`.
+            - "posterior_mean": A `batch_shape x num_pareto_samples x q x 1 x M`-dim
+                Tensor containing the posterior mean at the input `X`.
+            - "posterior_variance": A `batch_shape x num_pareto_samples x q x 1 x
+                M`-dim Tensor containing the posterior variance at the input `X`
+                excluding the observation noise.
+            - "observation_noise": A `batch_shape x num_pareto_samples x q x 1 x M`
+                -dim Tensor containing the observation noise at the input `X`.
+            - "posterior_with_noise": The posterior distribution at `X` which
+                includes the observation noise. This is used to compute the marginal
+                log-probabilities with respect to `p(y| x, D_n)` for `x` in `X`.
+        """
+        tkwargs = {"dtype": X.dtype, "device": X.device}
+        CLAMP_LB = torch.finfo(tkwargs["dtype"]).eps
+
+        posterior_statistics = dict()
+        # Compute the initial entropy term depending on `X`.
+        # TODO: Below we compute posterior_plus_noise twice:
+        #  (1) Firstly, we compute p(Y| X, D_n) when computing the initial entropy
+        #  (2) Secondly, we compute p(y| x, D_n) for x in X in order to compute
+        #  log(p(y|x, D_n)) for x in X in the Monte Carlo estimate..
+        #  This could be simplified if we could evaluate log(p(y|x, D_n)) using the
+        #  the posterior p(Y| X, D_n)
+        posterior_plus_noise = self.initial_model.posterior(X, observation_noise=True)
+
+        # Additional constant term.
+        add_term = (
+            0.5
+            * self.model.num_outputs
+            * (1 + torch.log(2 * pi * torch.ones(1, **tkwargs)))
+        )
+        # The variance initially has shape `batch_shape x (q*M) x (q*M)`
+        # prior_entropy has shape `batch_shape x num_fantasies`
+        initial_entropy = add_term + 0.5 * torch.logdet(
+            posterior_plus_noise.mvn.covariance_matrix
+        )
+        posterior_statistics["initial_entropy"] = initial_entropy
```

Comment:
```suggestion
        posterior_statistics = {"initial_entropy": initial_entropy}
```
A Meta-internal linter complains about the `dict()` call, and also let's define `posterior_statistics` closer to where it's used